frontend/javascripts/dashboard/dataset/dataset_cache_provider.js

```diff
@@ -89,8 +91,34 @@ export default function DatasetCacheProvider({ children }: { children: Node }) {
       setIsLoading(false);
     }
   }
+
+  async function updateDataset(datasetId: APIDatasetId) {
+    if (isLoading) return;
+    try {
+      setIsLoading(true);
+      const updateIdx = datasets.findIndex(
+        dataset =>
+          dataset.name === datasetId.name &&
+          dataset.owningOrganization === datasetId.owningOrganization,
+      );
+      if (updateIdx !== -1) {
+        const updatedDataset = await getDataset(datasetId);
+        // TODO is this bad practice?
```

Comment:
One open question would be, if it is fine to reuse the `datasets` array.

Reply:
No, this is considered bad practice as you suspected. In short, mutating state might lead to missed rendering updates because the object reference stays the same (see [this post](https://stackoverflow.com/questions/55178507/mutating-state-from-reacts-usestate-hook), for example).

You could simply do
```javascript
const newDatasets = datasets.slice();
newDatasets[updateIdx] = updatedDataset;
setDatasets(newDatasets);
```